Code/Data_science_project_exploration.Rmd

```diff
@@ -0,0 +1,280 @@
+
+create boxplot by year/month/day for total demand
+
+Get max/min timegap between each measure for all 3 datasets <- data integrity check.
+
+Check for seasonality trend for total demand, temperature
+
+create line chart to display actual - final to confirm model accuracy of aemo
+
+```{r data import and clean}
+##need to change to full outer join
+data <- totaldemand_nsw %>% full_join(temperature_nsw,by=c('DATETIME')) %>% select(-LOCATION,-REGIONID);
+
+##filter rows that do not have actual demand
+data <- data %>% filter(!is.na(TOTALDEMAND))
+
+##arrange by date in ascending
+data <- data %>% arrange(DATETIME);
+
+##fill in missing temperature (down fill)
+data <- data %>% fill(TEMPERATURE)
+
+##create table with period of initial forecast
+initial_forecast <- forecastdemand_nsw %>% 
+  filter(PERIODID==1) %>% 
+  select(DATETIME,
+         FORECASTDEMAND)%>%
+  rename('INITIAL_FORECAST'='FORECASTDEMAND')
+
+##create table with period of final forecast
+final_forecast_period <- forecastdemand_nsw %>% 
+  group_by(DATETIME) %>% 
+  summarise_at(vars(PERIODID),
+               list(PERIODID = max))
+final_forecast <- forecastdemand_nsw %>%
+  inner_join(final_forecast_period,by=c('DATETIME','PERIODID')) %>%
+  select(DATETIME,
+         FORECASTDEMAND) %>%
+  rename('FINAL_FORECAST'='FORECASTDEMAND')
+
+##join initial and final forecast to table data
+data <- data %>% 
+  left_join(initial_forecast,by=c('DATETIME')) %>%
+  left_join(final_forecast,by=c('DATETIME'))
+
+##fill in missing initial/final forecast demand (down fill)
+data <- data %>% 
+  fill(INITIAL_FORECAST) %>%
+  fill(FINAL_FORECAST)
+
+##Create DATE and MONTHDATE field for plotting
+data$DATE <- as.Date(substr(data$DATETIME,1,10))
+data$YEARMONTH<- as.Date(paste(substr(data$DATE,1,7), "-01", sep=""))
+  
+##convert DATETIME to date
+data$DATETIME <- ymd_hms(data$DATETIME)
+
+##remove unneeded table
+rm('final_forecast_period')
+
+```
+
+Generate relevant date time derivations.
+
+```{r create date fields}
+##create year/month/day/hour/minute fields
+data <- data %>% mutate(YEAR=year(DATETIME), 
+                        MONTH=month(DATETIME),
+                        DAY=day(DATETIME),
+                        HOUR=hour(DATETIME), 
+                        MINUTE=minute(DATETIME))
+
+##season
+data <- data %>%mutate(SEASON = case_when(
+  MONTH %in%  9:11 ~ "SPRING",
+  MONTH %in%  c(12, 1, 2)  ~ "SUMMER",
+  MONTH %in%  3:5  ~ "AUTUMN",
+  TRUE ~ "WINTER"))
+##create boolean for 8am~8pm timeslot
+data <- data %>%mutate(timeframe_8amto8pm = case_when(
+  HOUR %in%  8:20 ~ TRUE,
+  TRUE ~ FALSE))
+##Get days of week, monday = 1,...,sunday=7
+data<- data%>%mutate(DaysOfWeek=wday(DATETIME,
+                                     week_start = getOption("lubridate.week.start", 1)))
+##get weekday&weekend
+data <- data %>%mutate(WEEKDAYWEEKEND = case_when(
+  DaysOfWeek %in%  1:5 ~ "WEEKDAY",
+  DaysOfWeek %in%  6:7  ~ "WEEKEND",
+))
+
+##Deduplicate
+data <- data %>% distinct()
```

Comment:
is there duplicate data from above steps?

Reply:
There are 39 duplicates in the joined dataset (identified via creating a column for rolling datetime difference comparing row n and n-1, duplicates are captured by 0 differences). A few of these rows were sampled and they most definitely are duplicated rows.